modules/dists/CyclicDist.chpl

```diff
@@ -183,7 +183,105 @@ where the updates are made.
+            dataParTasksPerLocale=getDataParTasksPerLocale(),
+            dataParIgnoreRunningTasks=getDataParIgnoreRunningTasks(),
+            dataParMinGranularity=getDataParMinGranularity(),
+            param rank = _determineRankFromStartIdx(startIdx),
+            type idxType = _determineIdxTypeFromStartIdx(startIdx))
+    where isTuple(startIdx) || isIntegral(startIdx)
+  {
+    const value = new unmanaged CyclicImpl(startIdx, targetLocales,
+                                           dataParTasksPerLocale,
+                                           dataParIgnoreRunningTasks,
+                                           dataParMinGranularity,
+                                           rank, idxType
+                                          );
+    this.rank = rank;
+    this.idxType = idxType;
+
+    this.chpl_distHelp = new chpl_PrivatizedDistHelper(
+                          if _isPrivatized(value) then _newPrivatizedClass(value) else nullPid,
+                                                       value);
+  }
+
+    proc init(_pid : int, _instance, _unowned : bool) {
+      this.rank = _instance.rank;
+      this.idxType = _instance.idxType;
+
+      this.chpl_distHelp = new chpl_PrivatizedDistHelper(_pid,
+                                                         _instance,
+                                                         _unowned);
+    }
+
+    proc init(value) {
+      this.rank = value.rank;
+      this.idxType = value.idxType;
+
+      this.chpl_distHelp = new chpl_PrivatizedDistHelper(
+                                                         if _isPrivatized(value) then _newPrivatizedClass(value) else nullPid,
+                                                         _to_unmanaged(value));
+    }
+
+    // Note: This does not handle the case where the desired type of 'this'
+    // does not match the type of 'other'. That case is handled by the compiler
+    // via coercions.
+    proc init=(const ref other : Cyclic(?)) {
+      this.init(other._value.dsiClone());
+    }
+
+    proc clone() {
+      return new Cyclic(this._value.dsiClone());
+    }
+
+  @chpldoc.nodoc
+  inline operator ==(d1: Cyclic(?), d2: Cyclic(?)) {
+    if (d1._value == d2._value) then
+      return true;
+    return d1._value.dsiEqualDMaps(d2._value);
+  }
+
+  @chpldoc.nodoc
+  inline operator !=(d1: Cyclic(?), d2: Cyclic(?)) {
+    return !(d1 == d2);
+  }
+
+  proc writeThis(x) {
+    chpl_distHelp.writeThis(x);
+  }
+}
+
+
+@chpldoc.nodoc
+@unstable(category="experimental", reason="assignment between distributions is currently unstable due to lack of testing")
+operator =(ref a: Cyclic(?), b: Cyclic(?)) {
+  if a._value == nil {
+    __primitive("move", a, chpl__autoCopy(b.clone(), definedConst=false));
+  } else {
+    if a._value.type != b._value.type then
+      compilerError("type mismatch in distribution assignment");
+    if a._value == b._value {
+      // do nothing
+    } else
+        a._value.dsiAssign(b._value);
```

Comment:
Was this "do nothing" branch added to make things more readable? It seems just using `if a._value != b._value then a._value.dsiAssign(b._value);` or something like that would be simpler; this empty branch seems odd to me.

Reply:
I don't know, as this is just code that I've been copying over from the existing `_distribution` type, specializing it for Block and Cyclic without modifications other than the type signatures:  https://github.com/chapel-lang/chapel/blob/main/modules/internal/ChapelArray.chpl#L2300

Looking at the PR that added it, it appears that this branch used to contain some code and that Michael may have replaced the code with the comment to minimize the diff / retain the previous control flow structure:  https://github.com/chapel-lang/chapel/pull/4762/files#diff-52d2477eb48a33d9e81b64d64d74a50145a1bacdd880825e267b5bd5549559afR2964